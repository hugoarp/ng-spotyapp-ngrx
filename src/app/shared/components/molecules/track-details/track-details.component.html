<section class="track-details">
  <app-cover-img [cover]="cover"></app-cover-img>
  <app-text-b [text]="title"></app-text-b>
  <app-text-a [text]="description"></app-text-a>
</section>
