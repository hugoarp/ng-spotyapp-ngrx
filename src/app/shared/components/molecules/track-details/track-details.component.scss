@import "variables";

.track-details {
  display: flex;
  flex-direction: column;
  background-color: $dark-secondary;
  padding: 1rem;
  border-radius: 0.5rem;
  &:hover {
    background-color: $dark-tertiary;
  }
  &__footer {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
  }
  &__fav {
    cursor: pointer;
    font-size: 20px;
    &:hover {
      scale: 1.2;
    }
  }
}