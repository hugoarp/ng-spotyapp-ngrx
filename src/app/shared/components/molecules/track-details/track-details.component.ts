import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'app-track-details',
  templateUrl: './track-details.component.html',
  styleUrls: ['./track-details.component.scss'],
})
export class TrackDetailsComponent implements OnInit {
  @Input() cover: string = '';
  @Input() title: string = '';
  @Input() description: string = '';

  constructor() {}

  ngOnInit(): void {}
}
