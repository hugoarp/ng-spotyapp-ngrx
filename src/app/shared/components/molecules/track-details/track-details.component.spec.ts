import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TrackDetailsComponent } from './track-details.component';

describe('TrackDetailsComponent', () => {
  let component: TrackDetailsComponent;
  let fixture: ComponentFixture<TrackDetailsComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [TrackDetailsComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(TrackDetailsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('debe emitir el evento favoriteEvent con el track al hacer click en botÃ³n de favorito', () => {
    const spy = jest.spyOn(component.favoriteEvent, 'emit');
    const button = fixture.nativeElement.querySelector('i');
    button.click();
    expect(spy).toHaveBeenCalledWith(component.track);
  });
});
