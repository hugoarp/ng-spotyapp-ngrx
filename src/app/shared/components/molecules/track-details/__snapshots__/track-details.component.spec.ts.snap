// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackDetailsComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <section
    class="track-details"
  >
    <app-cover-img />
    <app-text-b
      class="track-details__text"
    />
    <article
      class="track-details__footer"
    >
      <app-text-a
        class="track-details__text"
      />
      <i
        class="material-icons track-details__fav"
      >
        favorite 
      </i>
    </article>
  </section>
</div>
`;
